docs/BackupAndRestore.md

```diff
@@ -0,0 +1,594 @@
+
+```bash
+kubectl logs full-hotcopy-demo-cronjob-1613663760-r8lm8
+Starting full backup for database demo on processes with labels 'backup cluster0 ' ...
+'backup database' failed: No SMs found matching the provided labels backup cluster0
+Error running hotcopy 1
+```
+
+Further investigation shows that the HC SMs statefulset has been scaled down to 0 replicas.
+
+```bash
+kubectl get statefulsets.apps
+NAME                                      READY   AGE
+admin-nuodb-cluster0                      1/1     3h41m
+sm-database-nuodb-cluster0-demo           2/2     3h41m
+sm-database-nuodb-cluster0-demo-hotcopy   0/0     3h41m
+```
+
+> **IMPORTANT**: It is important to take periodic journal hot copies when journal hot copy is enabled. Otherwise, the journal will keep growing which can lead to filling the archive volume space.
+
+It is possible to disable backup jobs creation by setting `hotcopy.enableBackups` to `false`.
+Custom backup jobs that use the `nuobackup` script can be configured to support more complex workflows and backup strategies.
+
+### Restore/Import source and type
+
+The restore/import source location can be one of the following:
+- remote _URL_ in a form of `protocol://authority/path`.
+The restore logic uses [cURL](https://curl.se/) to download a stream of data into an SM pod and restore the archive from it.
+Standard protocols such as _HTTP/s_, _FTP_ and _SFTP_ can be used as a source.
+Any other protocols that have built-in support in the shipped `cURL` binary in the NuoDB image can be used as well.
+Only basic authentication can be specified for remote sources. A remote location that has an authentication token embedded into the _URL_ can also be used.
+The remote URL should point to the `tar.gz` file which will be automatically extracted using the configured `stripLevels` setting (by default `1`). 
+These are the number of leading path elements removed by the `tar` command during archive extraction.
+
+- backupset available in the backup root directory.
+HC SMs have a backup volume attached and are selected for hot copy operation during the scheduled backups. Only these SMs have access to local hot copies and their archives can be restored using a local backup set.
+
+The most recent successful backup set name can be referenced in one of the following ways:
+- `:latest` - The latest backup set name among all backup groups.
+- `:group-latest` - The latest backup set name for the backup group in current database deployment.
+
+Two restore/import source types are available:
+- _stream_ - Used as a synonym for a copy of NuoDB archive which contains both the archive and journal directories.
+A remote _stream_ source is downloaded and extracted directly into the archive directory which doesn't require extra space for holding the restore/import source file.
+- _backupset_ - Hot copy backup set which is either fetched from a remote source or available in the backup directory.
```

Comment:
like the previous line, explain this requires additional space for download/extract?